error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ImplicitBlockingFlux.java

```diff
@@ -0,0 +1,106 @@
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.suppliers.Supplier;
+import com.google.errorprone.suppliers.Suppliers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.code.Types;
+import com.sun.tools.javac.util.Position;
+import tech.picnic.errorprone.bugpatterns.util.ThirdPartyLibrary;
+
+/**
+ * A {@link BugChecker} that flags {@link reactor.core.publisher.Flux} operators that are implicitly
+ * blocking.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary = "Avoid using `Flux` operators that implicitly block",
+    link = BUG_PATTERNS_BASE_URL + "ImplicitBlockingFlux",
+    linkType = CUSTOM,
+    severity = SUGGESTION,
+    tags = STYLE)
+public final class ImplicitBlockingFlux extends BugChecker implements MethodInvocationTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Matcher<ExpressionTree> FLUX_WITH_IMPLICIT_BLOCK =
+      instanceMethod()
+          .onDescendantOf(Suppliers.typeFromString("reactor.core.publisher.Flux"))
+          .namedAnyOf("toIterable", "toStream")
+          .withNoParameters();
+  private static final Supplier<Type> STREAM = Suppliers.typeFromString("java.util.stream.Stream");
+
+  /** Instantiates a new {@link ImplicitBlockingFlux} instance. */
+  public ImplicitBlockingFlux() {}
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    if (!FLUX_WITH_IMPLICIT_BLOCK.matches(tree, state)) {
+      return Description.NO_MATCH;
+    }
+
+    Description.Builder description = buildDescription(tree);
+
+    description.addFix(SuggestedFixes.addSuppressWarnings(state, canonicalName()));
+    if (ThirdPartyLibrary.GUAVA.isIntroductionAllowed(state)) {
+      description.addFix(
+          trySuggestFix("com.google.common.collect.ImmutableList.toImmutableList", tree, state));
+    }
+    description.addFix(trySuggestFix("java.util.stream.Collectors.toList", tree, state));
+
+    return description.build();
+  }
+
+  private static SuggestedFix trySuggestFix(
+      String fullyQualifiedMethodInvocation, MethodInvocationTree tree, VisitorState state) {
+    SuggestedFix.Builder importFix = SuggestedFix.builder();
+    String replacement =
+        SuggestedFixes.qualifyStaticImport(fullyQualifiedMethodInvocation, importFix, state);
+
+    return replaceWithExplicitBlock(tree, replacement + "()", importFix.build(), state);
+  }
+
+  private static SuggestedFix replaceWithExplicitBlock(
+      MethodInvocationTree tree,
+      String collectArgument,
+      SuggestedFix importFix,
+      VisitorState state) {
+    String collect = String.format("collect(%s)", collectArgument);
+    Types types = state.getTypes();
+    String explicitBlock =
+        types.isSubtype(ASTHelpers.getResultType(tree), types.erasure(STREAM.get(state)))
```

Comment:
```suggestion
        ASTHelpers.isSubtype(ASTHelpers.getResultType(tree), STREAM.get(state), state)
```
(Also performs type erasure.)